solutions/devsprint-matheus-reis-1/RealEstateApp/UI/Components/ButtonView.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  ButtonView.swift
+//  RealEstateApp
+//
+//  Created by Cora on 12/05/22.
+//
+
+import Foundation
+import UIKit
+
+struct ButtonViewConfiguration {
+    var buttonText: String
+}
+
+class ButtonView: UIButton, ViewCode {
+    
+    init() {
+        super.init(frame: .zero)
+        setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func updateView(with configuration: ButtonViewConfiguration) {
```

Comment:
Esse método "limita" a definir o título do botão apenas para o estado `.normal`. Como esse componente é uma extensão do UIButton, quem for utilizá-lo poderá usar o próprio método `setTitle`. Então, esse método não é mais necessário, e no teste você pode usar só o setTitle mesmo